### PR TITLE
Feature/Windows support upgrade

### PR DESCRIPTION
**Pull request by awsitcloudpro**
*Issue #, if available:*
aws-ia/terraform-ssp-amazon-eks#89

*Description of changes:*
[EKS Windows support mechanism](https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html#enable-windows-support) was recently upgraded. The upgraded VPC CNI plugin no longer needs any VPC controller resources in the data plane. Updated this repo accordingly to replace the Windows VPC controller Helm chart with `amazon-vpc-cni` configmap.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: modules/aws-eks-self-managed-node-groups/README.md [hunk 1]
```diff
@@ -13,9 +13,10 @@ This module allows you to create on-demand or spot self managed Linux or Windows
   #---------------------------------------------------------#
 
   # -------- START WINDOWS ONLY
-  enable_windows_support                    = false
-  windows_vpc_resource_controller_image_tag = "v0.2.7" # enable_windows_support= true
-  windows_vpc_admission_webhook_image_tag   = "v0.2.7" # enable_windows_support= true
+  enable_windows_support                    = false # Set to true to enable Windows support
+  # Ensure AWS VPC CNI is used
+  enable_vpc_cni_addon  = true
```
[the remark this answers is not in the document]
Comment (awsitcloudpro) [reply modules/aws-eks-self-managed-node-groups/README.md:18]: This example is for self managed node groups. Windows nodegroups are a subset of that use case. That's why the default setting in this example is to just enable Linux self-managed nodegroups, and the comment indicates how one can enable Windows support in addition to that.

File: main.tf [hunk 2]
```diff
@@ -110,3 +110,18 @@ module "emr_on_eks" {
...
+  }
+
+  data = {
+    "enable-windows-ipam" = var.enable_windows_support ? "true" : "false"
+  }
+
+  depends_on = [
+    module.aws_eks
+  ]
+}
```
[the remark this answers is not in the document]
Comment (awsitcloudpro) [reply main.tf:127]: It is automatically attached when using the [Terraform EKS](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/6959b9bae32309357bc97a85a1f09c7b590c8a6d/main.tf#L155-L160) module, as this repo does.

**Review by kcoleman731 — COMMENTED** [review]
LGTM needs docs.

**Review by vara-bonthu — APPROVED** [review]
LGTM

**Pull request merged**